k2/csrc/intersect_pruned.cu

```diff
@@ -730,81 +735,84 @@ class MultiGraphDenseIntersectPruned {
-        int32_t state_idx01 =
-           state_map_acc(fsa_id, dest_a_fsas_state_idx01);
-        info.u.dest_info_state_idx01 = state_idx01;
-
-        // state_idx01 will be -1 for states that were pruned out because
-        // they were outside the beam.
-        if (state_idx01 < 0)
-          return;
-        // multiple threads may write the same value to the address written to
-        // in the next line.
-        kept_states_data[state_idx01].a_fsas_state_idx01 =
-            dest_a_fsas_state_idx01;
-        int32_t end_loglike_int = FloatToOrderedInt(info.end_loglike);
-        // Set the forward log-like of the dest state to the largest of any of
-        // those of the incoming arcs.  Note: we initialized this in
-        // lambda_init_loglike above.
-        atomicMax(&(kept_states_data[state_idx01].forward_loglike),
-                  end_loglike_int);
-      };
-      Eval(c_, arc_info.NumElements(), lambda_set_arcs_and_states);
+      K2_EVAL(
+          c_, arc_info.NumElements(), lambda_set_arcs_and_states,
+          (int32_t arc_idx012)->void {
+            int32_t fsa_id = ai_row_ids1[ai_row_ids2[arc_idx012]];
+
+            ArcInfo &info = ai_data[arc_idx012];
+
+            int32_t dest_a_fsas_state_idx01 = info.u.dest_a_fsas_state_idx01;
+            // state_idx01 is the index into ans->states, of the destination
+            // state. Note: multiple arcs may enter this state, which is why we
+            // had to set that in a separate kernel (lambda_modify_state_map).
+            int32_t state_idx01 =
+                state_map_acc(fsa_id, dest_a_fsas_state_idx01);
+            info.u.dest_info_state_idx01 = state_idx01;
+
+            // state_idx01 will be -1 for states that were pruned out because
+            // they were outside the beam.
+            if (state_idx01 < 0) return;
+            // multiple threads may write the same value to the address written
+            // to in the next line.
+            kept_states_data[state_idx01].a_fsas_state_idx01 =
+                dest_a_fsas_state_idx01;
+            int32_t end_loglike_int = FloatToOrderedInt(info.end_loglike);
+            // Set the forward log-like of the dest state to the largest of any
+            // of those of the incoming arcs.  Note: we initialized this in
+            // lambda_init_loglike above.
+            atomicMax(&(kept_states_data[state_idx01].forward_loglike),
+                      end_loglike_int);
+          });
     }
     {
       NVTX_RANGE("LambdaResetStateMap");
       const int32_t *next_states_row_ids1 = ans->states.shape.RowIds(1).Data();
-      auto lambda_reset_state_map =
-          [=] __host__ __device__(int32_t state_idx01) -> void {
-        int32_t a_fsas_state_idx01 =
-           kept_states_data[state_idx01].a_fsas_state_idx01,
-           fsa_idx0 = next_states_row_ids1[state_idx01];
-        K2_CHECK_EQ(state_map_acc(fsa_idx0, a_fsas_state_idx01), state_idx01);
```

Comment:
I changed this CHECK to **D**CHCECK as it is in a lambda, i.e., inside a `for` loop.